string name
uint8 inst_num
uint8 interface_num
uint8 device_num